src/sqlfluff/rules/L059.py

```diff
@@ -84,7 +84,13 @@ def _eval(self, context: RuleContext) -> Optional[LintResult]:
             identifier_contents = context.segment.raw[1:-1]
 
         # Ignore the segments that are not of the same type as the defined policy above.
-        if context.segment.name not in context_policy:
+        # Also TSQL has a keyword called QUOTED_IDENTIFIER which maps to the name so
+        # need to explicity check for that.
+        if (
+            context.segment.name not in context_policy
+            or context.segment.raw.lower() == "quoted_identifier"
+            or context.segment.raw.lower() == "naked_identifier"
```

Comment:
```suggestion
            or context.segment.raw.lower() in ("quoted_identifier", "naked_identifier")
```

Reply:
I'm wondering about this code -- it's checking `raw`, but raw is the actual, rendered code. I'm surprised this isn't `name` or `type`.

Reply:
The problem is for a keyword the name is the same as the raw. Hence this issue.

We could check the name and if it’s a keyword, but thought above check was more comprehensive.

Reply:
BTW the full `if` statement is:

```py
        if (
            context.segment.name not in context_policy
            or context.segment.raw.lower() in ("quoted_identifier", "naked_identifier")
        ):
```

That is, if the name is not one we're interested in OR the raw value is literally the value `QUOTED_IDENTIFIER` (which is a `name` in SQLFluff parser language BUT ALSO an actual KEYWORD in T-SQL 😱), then it's not one we care about.